src/components/Field.vue

```diff
@@ -43,9 +52,39 @@ export default {
       }
 
       return this.value;
+    },
+    onSubmit() {
+      if (!this.provider) {
+        return;
```

Comment:
To keep the return type consistent

```suggestion
        return Promise.resolve();
```

Reply:
This is not required: `Promise.all([undefined, Promise.resolved()])` is perfectly acceptable. This function either returns undefined or a promise, I don't see any problem with that.